### DISCUSSION TITLE
Value type filter using JMESPath on JSON array

### DISCUSSION THREAD

**Discussion by farhanmkhan** [cloud-custodian/cloud-custodian#7343] [General]
Hi,

I am trying to write a filter to get all Security Groups which have the IP Protocol as "tcp". Following works very well.

    filters:
      - type: value
        key: "IpPermissions[0].IpProtocol"
        value: "tcp"
        op: equal

Above filter will only check the 0th element. I'd like to do the same check on all the elements of  IpPermissions array. I tried following but didn't work:

    filters:
      - type: value
        key: "IpPermissions[*].IpProtocol"
        value: "tcp"
        op: equal

neither did following work

    filters:
      - type: value
        key: "IpPermissions[].IpProtocol"
        value: "tcp"
        op: equal

or even sliding didn't work

    filters:
      - type: value
        key: "IpPermissions[0:1].IpProtocol"
        value: "tcp"
        op: equal

Please advise!

Thanks

**Comment by kapilt** [comment]
use the ingress/egress filter which support this much more nicely and deal with various other details of security group permissions.

See `custodian schema aws.security-group.filters.ingress`

**Comment by AbhishekSinha47** [comment]
Try this, it would work:
   ``` 
filters:
     - type: value
        key: IpPermissions[].IpProtocol
        value_type: swap
        op: in
        value: tcp
```
